ddclient.in

```diff
@@ -4238,6 +4259,174 @@ sub nic_dnsexit_update {
+        my %status = (
+             '0'        => [ 'good',    'Success! Actions got executed successfully.' ],
+             '1'        => [ 'warning', 'Some execution problems. May not indicate actions failures. Some action may got executed fine and some may have problems.' ],
+             '2'        => [ 'badauth', 'API Key Authentication Error. The API Key is missing or wrong.' ],
+             '3'        => [ 'error',   'Missing Required Definitions. Your JSON file may missing some required definitions.' ],
+             '4'        => [ 'error',   'JSON Data Syntax Error. Your JSON file has syntax error.' ],
+             '5'        => [ 'error',   'JSON Defined Record Type not Supported. Your JSON may try to update some record type not supported by our system.' ],
+             '6'        => [ 'error',   'System Error. Our system problem. May not be your problem. Contact our support if you got such error.' ],
+             '7'        => [ 'error',   'Error getting post data. Our server has problem to receive your JSON posting.' ],
+        );
+        my $ip = delete $config{$h}{'wantip'};
+        info("Going to update IP address to %s for %s.", $ip, $h);
+        # Set the URL of the API endpoint
+        my $url = "https://$config{$h}{'server'}$config{$h}{'path'}";
+
+        # Set JSON payload
+        my $data = encode_json({
+                        apikey => $config{$h}{'password'},
+                        domain => $h,
+                        update => {
+                           type => $config{$h}{'record-type'},
+                           name => $h,
+                           content => $ip,
+                           ttl => $config{$h}{'ttl'}},
+                    });
+
+        # Set additional headers
+        my $header = "Content-Type: application/json\n";
+        $header .= "Accept: application/json";
+
+        # Make the call
+        my $reply = geturl(
+            proxy     => opt('proxy'),
+            url       => $url,
+            headers   => $header,
+            method    => 'POST',
+            data      => $data,
+        );
+
+        # No reply, declare as failed
+        if (!defined($reply) || !$reply) {
+            failed("updating %s: Could not connect to %s%s.", $h, $config{$h}{'server'}, $config{$h}{'path'});
+            $config{$h}{'status'} = 'failed';
+            last;
+        };
+
+        # Reply found
+        debug("%s", $reply);
+        # $ok is mandatory?
+        my $ok = header_ok($h, $reply);
+
+        # Extract the HTTP response code
+        (my $http_status) = ($reply =~ m%^s*HTTP/.*\s+(\d+)%i);
+        debug("HTTP response code: %s", $http_status);
+
+        # If not 200, bail
+        if ( $http_status != "200"){
+            failed("==Failed to update Host==\n%s to IP:%s", $h, $ip);
+            failed("==HTTP response code==\n%s", $http_status);
+            failed("==Full reply==\n%s", $reply) unless opt('verbose');
+            $config{$h}{'status'} = 'failed';
```

Comment:
Can we remove the == == from the logs? Or is this the default in other providers?

Reply:
Done